### PR TITLE
GitHub release stats

### PR DESCRIPTION
**Pull request by richsilv**
Adds function to calculate aggregate stats (number of commits, lines added, lines removed) for the supplied repos and a given version number.

Stats are calculated on the basis of comparison to the previous version, or else the whole commit history if this is the first tag.

Also adds a CLI example, which can be run with:

```sh
> node example/github-release-stats-cli.js --repo olizilla/tags --version v2.0.0
```

### CODE REVIEW

**Comment by alanshaw** [comment]
👓

File: example/github-release-stats-cli.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+github-release-stats-cli
+
+Make sure you've got a github.rc in the project root with your username and an api key.
```
Comment (alanshaw) [inline example/github-release-stats-cli.js]: .githubrc

File: lib/github-interface.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      additions: memo.additions + file.additions,
+      deletions: memo.deletions + file.deletions
+    }
+  }, { additions: 0, deletions: 0 })
+
+  return {
+    commits: stats.commits.length,
+    additions: lineStats.additions,
+    deletions: lineStats.deletions
+  }
```
Comment (alanshaw) [inline lib/github-interface.js:86]: Just comparing this with the [pulse page](https://github.com/tableflip/study-bundles-app/pulse), can we return the number of files changed, PRs merged and issues closed?

**Review by alanshaw — CHANGES_REQUESTED** [review]
I got the following error:
`Error: Cannot find module 'mime-types'`

Removing `package-lock.json` and re-installing solved it, but left me with a `package-lock.json` with many changes from the original.

File: lib/github-interface.js (hunk 2 above)
Comment (bmordan) [reply lib/github-interface.js:86]: Lets do this later it is quite tricky

**Pull request merged**